hf/hf.go

```diff
@@ -122,13 +127,12 @@ func (e *HuggingFaceEmbeddingFunction) CreateEmbedding(documents []string) ([][]
 	return response.Embeddings, nil
 }
 
-func (e *HuggingFaceEmbeddingFunction) CreateEmbeddingWithModel(documents []string, model string) ([][]float32, error) {
-	e.apiClient.Model = model
+func (e *HuggingFaceEmbeddingFunction) EmbedQuery(ctx context.Context, document string) ([]float32, error) {
 	response, err := e.apiClient.CreateEmbedding(&CreateEmbeddingRequest{
```

Comment:
let's also pass the context to CreateEmbedding and then pass it to the HTTP request